arduino-ide-extension/src/browser/local-cache/local-cache-fs-provider.ts

```diff
@@ -88,7 +88,10 @@ export class LocalCacheFsProvider
   }
 
   protected async init(fileService: FileService): Promise<void> {
-    const config = await this.configService.getConfiguration();
+    const { config, messages } = await this.configService.config();
+    if (!config) {
+      throw new Error(messages ? messages[0] : 'cli config was undefined');
```

Comment:
check the typings here. increase TS level? if config is not defined, messages must be defined

Reply:
> local cache provider must re-init on data dir config change

WONTDO: there is no way in IDE2 to change the data dir. Hence, handling this change is outside of the scope of the PR.